tests/validation_tests/specific_validation_tests/pathfinders/test_round_1.py

```diff
@@ -218,6 +219,52 @@ def test__check_credible_plan_fields_fails(mock_df_dict):
     ]
 
 
+def test__check_current_underspend_passes(mock_df_dict):
+    original_reporting_period = mock_df_dict["Reporting period"].iloc[0, 0]
+    original_current_underspend = mock_df_dict["Current underspend"].iloc[0, 0]
+
+    # Test case where reporting period is Q4 and current underspend is entered
+    assert original_reporting_period == "Q4 2023/24: Jan 2024 - Mar 2024"
+    assert original_current_underspend == 0
+    error_messages = _check_current_underspend(mock_df_dict)
+    assert error_messages == []
+
+    # Test case where reporting period is Q4 and current underspend is missing
+    mock_df_dict["Current underspend"].iloc[0, 0] = pd.NA
+    error_messages = _check_current_underspend(mock_df_dict)
+    assert error_messages == []
+    mock_df_dict["Current underspend"].iloc[0, 0] = original_current_underspend  # Reset the mock data
+
+    # Test case where reporting period is not Q4 and current underspend is entered
+    mock_df_dict["Reporting period"].iloc[0, 0] = "Q1 2024/25: Apr 2024 - Jun 2024"
+    error_messages = _check_current_underspend(mock_df_dict)
+    assert error_messages == []
+    mock_df_dict["Reporting period"].iloc[0, 0] = original_reporting_period  # Reset the mock data
+
+
+def test__check_current_underspend_fails(mock_df_dict):
+    original_reporting_period = mock_df_dict["Reporting period"].iloc[0, 0]
+    original_underspend = mock_df_dict["Current underspend"].iloc[0, 0]
+
+    # Test case where reporting period is not Q4 and current underspend is missing
+    mock_df_dict["Reporting period"].iloc[0, 0] = "Q1 2024/25: Apr 2024 - Jun 2024"
+    mock_df_dict["Current underspend"].iloc[0, 0] = pd.NA
+    error_messages = _check_current_underspend(mock_df_dict)
+    assert error_messages == [
+        Message(
+            sheet="Finances",
+            section="Current underspend",
+            cell_indexes=("B1",),
+            description="Current underspend must be filled in if the reporting period is not Q4.",
+            error_type=None,
+        )
+    ]
+
+    # Reset the mock data
+    mock_df_dict["Reporting period"].iloc[0, 0] = original_reporting_period
+    mock_df_dict["Current underspend"].iloc[0, 0] = original_underspend
```

Comment:
Could we use a copy of this dict with the changes rather than mutating/resetting it here?

Reply:
`mock_df_dict` should probably just be function-scoped so that we don't even have to think about changes to it being carried between tests. That seems too painful to want to think about. Any reason we can't do this?

Reply:
Have just tried switching the fixture from module scope to function scope and all tests pass, so seems like a no-brainer. As this pattern of capturing original values from the mock_df_dict, mutating, running the function and then resetting to those values is prevalent throughout this entire module, not just these new functions, what do you think to using Gideon's copy technique on this PR in particular to keep the scope limited to the ticket, but then for me to create a separate ticket and PR to change the scope and remove all the capturing original values / resetting to those values logic?

Reply:
Whichever you prefer is fine with me :)